src/runtime/execution.rs

```diff
@@ -144,7 +144,42 @@ impl Execution {
         // Run a single step of the chosen task.
         let ret = match next_step {
             NextStep::Task(continuation) => {
-                panic::catch_unwind(panic::AssertUnwindSafe(|| continuation.borrow_mut().resume()))
+                // Enter the Task's span
+                ExecutionState::with(|state| {
+                    tracing::dispatcher::get_default(|subscriber| {
+                        let current_span_id = tracing::Span::current().id();
+                        if let Some(span_id) = current_span_id.as_ref() {
+                            subscriber.exit(span_id);
+                        }
+
+                        if let Some(span_id) = state.current().span.id().as_ref() {
+                            subscriber.enter(span_id)
+                        }
+
+                        if state.config.record_steps_in_span {
+                            state.current().step_span.record("i", state.current_schedule.len());
+                        }
+                    });
+                });
+
+                let result = panic::catch_unwind(panic::AssertUnwindSafe(|| continuation.borrow_mut().resume()));
+
+                // Leave the Task's span and store which span it exited in order to restore it the next time the Task is run
+                ExecutionState::with(|state| {
+                    tracing::dispatcher::get_default(|subscriber| {
+                        let current_span = tracing::Span::current();
+                        if let Some(span_id) = current_span.id().as_ref() {
+                            subscriber.exit(span_id);
+                        }
+                        state.current_mut().span = current_span;
```

Comment:
This makes me wonder if the field should be renamed from `span` to `last_span` ?